R/plot_step_ahead_model_output.R

```diff
@@ -0,0 +1,900 @@
+#'
+#' @importFrom plotly plot_ly
+simple_model_plot <- function(
+    plot_model, df_point, df_ribbon, plot_truth, truth_data, opacity = 0.25,
+    line_color = NULL, top_layer = "model_output", show_truth_legend = TRUE,
+    interactive = TRUE, fill_by = "model_id", x_col_name = "target_date",
+    x_truth_col_name = "time_idx", ...) {
+  # prerequisite
+  if (is.null(plot_model)) {
+    if (interactive) {
+      plot_model <- plotly::plot_ly()
+    } else {
+      plot_model <- ggplot2::ggplot()
+    }
+
+  }
+  arguments <- list(...)
+
+  if (top_layer == "model_output") {
+    if (interactive) {
+      # Truth Data
+      plot_model <- plotly_truth_data(plot_model, truth_data, plot_truth,
+                                      show_truth_legend, arguments,
+                                      x_col_name = x_truth_col_name)
+      # Projection data
+      plot_model <- plotly_proj_data(plot_model, df_point, df_ribbon,
+                                     line_color, opacity, arguments,
+                                     fill_by = fill_by, x_col_name = x_col_name)
+    } else {
+      # Truth Data
+      plot_model <- static_truth_data(plot_model, truth_data, plot_truth,
+                                      x_col_name = x_truth_col_name)
+      # Projection data
+      plot_model <- static_proj_data(plot_model, df_point, df_ribbon,
+                                     line_color, opacity, fill_by = fill_by,
+                                     x_col_name = x_col_name)
+    }
+
+  } else if (top_layer == "truth") {
+    if (interactive) {
+      # Projection data
+      plot_model <- plotly_proj_data(plot_model, df_point, df_ribbon,
+                                     line_color, opacity, arguments,
+                                     fill_by = fill_by, x_col_name = x_col_name)
+      # Truth Data
+      plot_model <- plotly_truth_data(plot_model, truth_data, plot_truth,
+                                      show_truth_legend, arguments,
+                                      x_col_name = x_truth_col_name)
+    } else {
+      # Projection data
+      plot_model <- static_proj_data(plot_model, df_point, df_ribbon,
+                                     line_color, opacity,
+                                     fill_by = fill_by, x_col_name = x_col_name)
+      # Truth Data
+      plot_model <- static_truth_data(plot_model, truth_data, plot_truth,
+                                      x_col_name = x_truth_col_name)
+    }
+  }
+
+  return(plot_model)
+}
+
+#' Plot projection data
+#'
+#' Use Plotly or ggplot2 to plot projection model output with or without
+#' truth data.
+#'
+#' @param all_plot a list with two data frame: one for plain lines,
+#' one for ribbons plotting (in a wide format)
+#' @param all_ens a list with two data frame: one for plain lines,
+#'  one for ribbons plotting (in a wide format) for a unique `model_id` value
+#'  associated with specific color (`ens_color`). NULL is no specific layout
+#'  required
```

Comment:
I'm struggling a little to understand the difference between `all_plot` and `all_ens`